pkgs/desktops/gnome-3/extensions/update-extensions.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -I nixpkgs=../../../.. -i python3 -p python3
+
+import json
+import urllib.request
+import urllib.error
+from typing import List, Dict, Optional, Any, Tuple
+import logging
+from operator import itemgetter
+
+logging.basicConfig(level=logging.DEBUG)
+
+# We don't want all those deprecated legacy extensions
+# Group extensions by Gnome "major" version for compatibility reasons
+supported_versions = {
+    "36": "3.36",
+    "38": "3.38",
+}
+
+
+def fetch_sha256sum(uuid: str, version: str) -> str:
+    """
+    Download the extension and hash it. We use `nix-prefetch-url` for this for efficiency reasons
+    """
+    import subprocess
```

Comment:
Move this to the top of the file. Imports shouldn't happen lazily further down the script.